File: Navtrack.Api.Model/Devices/DeviceModel.cs
using Navtrack.Api.Model.Custom;

namespace Navtrack.Api.Model.Devices
{
    public class DeviceModel : IModel
    {
        public int Id { get; set; }
        public int AssetId { get; set; }
        public string DeviceId { get; set; }
        public DeviceTypeModel DeviceType { get; set; }
        public int LocationsCount { get; set; }
        public bool IsActive { get; set; }
    }
}

File: Navtrack.Core/ClientApp/src/apis/types/device/DeviceModel.ts
import { DeviceTypeModel, DefaultDeviceTypeModel } from "./DeviceTypeModel";

export type DeviceModel = {
  id: number;
  assetId: number;
  deviceId: string;
  deviceType: DeviceTypeModel;
  locationsCount: number;
};

export const DefaultDeviceModel : DeviceModel = {
  id: 0,
  assetId: 0,
  deviceId: "",
  deviceType: DefaultDeviceTypeModel,
  locationsCount: 0
};